pkg/providers/pricing/suite_test.go

```diff
@@ -252,3 +258,34 @@ var _ = Describe("Pricing", func() {
 			To(ContainElements("Linux/UNIX", "Linux/UNIX (Amazon VPC)"))
 	})
 })
+
+func getPricingEstimateMetricValue(instanceType string, capacityType string, zone string) float64 {
+	gathering, err := crmetrics.Registry.Gather()
+	Expect(err).To(BeNil())
+	var value *float64
+	for _, mf := range gathering {
+		if mf.GetName() == "karpenter_pricing_instance_estimate" {
+			for _, m := range mf.GetMetric() {
+				instanceTypeMatch := false
+				capacityTypeMatch := false
+				zoneMatch := false
+				for _, label := range m.Label {
```

Comment:
Consideration: You should be able to simplify this for loop and the following if check with three `lo.Find()` calls. 
```suggestion
				it, itFound := lo.Find(m.Label, func(label labelType, _ int) (bool) {
				    return aws.StringValue(label.Name) == "node_kubernetes_io_instance_type"
				}
				...
				...
				
				if itFound && ctFound &&zoneFound  {
				    value = m.GetGauge().Value
				}
```

Reply:
Done. Good idea.